src/client.cc

```diff
@@ -169,7 +170,12 @@ DisplayLine Client::generate_mode_line() const
         HashMap<String, DisplayLine> atoms{{ "mode_info", context().client().input_handler().mode_line() },
                                            { "context_info", {generate_context_info(context()),
                                                               context().faces()["Information"]}}};
-        auto expanded = expand(modelinefmt, context(), ShellContext{},
+        const NormalParams* normal_params = context().client().input_handler().get_normal_params();
+        ShellContext shell_context = { {}, {
+        {"register", normal_params ? String(normal_params->reg) : String("") }, 
+        {"count", normal_params ? String(to_string(normal_params->count)) : ""}}};
```

Comment:
I would prefer to not have $kak_register and $kak_count defined if normal_params is nullptr, so more `auto shell_context = normal_params ? ShellContext{ ... } : ShellContext{};`

Reply:
The reason why I wanted to avoid this, is that it means you can only use it through a shell invocation, and not directly in the modeline, as it will give an invalid expansion error. Having the expansion always available, but empty, feels more user friendly to me than having a modelinefmt that errors in some modes

Reply:
Good point, fixed that.